src/zenml/integrations/kubeflow/orchestrators/kubeflow_orchestrator.py

```diff
@@ -508,14 +508,15 @@ def _get_environment_vars_from_secrets(
 
         Raises:
             ProvisioningError: If the stack has no secrets manager."""
-
+        environment_vars: Dict[str, str] = {}
         secret_manager = Repository().active_stack.secrets_manager
-        if not secret_manager:
+        if secrets and secret_manager:
+            for secret in secrets:
+                secret_schema = secret_manager.get_secret(secret)
+                environment_vars.update(secret_schema.content)
+        elif secrets and not secret_manager:
```

Comment:
What happens in the else case? This would be `if not secrets and not secret_manager` and `if not secrets and secret_manager`? :-) Should we make these explicit?

Reply:
I was a bit worried about the same thing! if there are no secrets what would be the best way to handle that.
I was thinking about these implementations but I don't know which one of them is the right one:

- Add else condition with a log that explains that no secrets were added to the pipeline. (this is for the users: but it could be a bit confusing rather than helpful)
- Add else condition with a docstring and pass statement. (This would be good for us, perhaps we could have some change in this condition in the future)

would love to know which one you think should be implemented

Reply:
FWIW, my vote would be for the second option. That way we have it explicit, but no needless logging. After all, if a user isn't passing in any secrets, and they don't have a secrets manager as part of their stack, then we probably don't need to tell them that 'no secrets manager was used and no secrets were passed in' etc.

Reply:
Yes, Id also go for option 2 but make it a comment not a docstring (thats what you meant I think anyway)